.github/scripts/sync-with-master.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash -eux
+
+#
+# We need these config parameters set in order to do the git-merge.
+#
+git config user.name "${GITHUB_ACTOR}"
+git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+
+#
+# We obtain the original name for the branch we'll be syncing with
+# master. We gather this information here, since we'll change the name
+# of the local branch later, and we'd like to use the original branch
+# name when opening the pull request.
+#
+BRANCH=$(git symbolic-ref --short HEAD)
+[[ -n "${BRANCH}" ]] || exit 1
+
+#
+# We need the full git repository history in order to do the git-merge.
+#
+git fetch --unshallow
+git merge origin/master
+
+#
+# We modify the local branch name and upstream tracking branch such that
+# the correct operations occur when pushing and opening the pull request.
+#
+git branch -M "sync/${BRANCH}"
```

Comment:
not a big deal but maybe change this to `sync-with-master/${BRANCH}` or something like that?

Reply:
done.